R/authors.R

```diff
@@ -0,0 +1,103 @@
+#'
+#' The `use_author()` function should be used after [create_package()] or [use_description()].
+#'
+#' If you create a lot of packages, consider storing personalized defaults as a
+#' named list in an option named `"usethis.description"`. [use_description()] will automatically
+#' fill using this information. Here's an example of code to include in `.Rprofile`:
+#'
+#' ```
+#' options(
+#'   usethis.description = list(
+#'     `Authors@R` = 'person("Jane", "Doe", email = "jane@example.com", role = c("aut", "cre"),
+#'                           comment = c(ORCID = "YOUR-ORCID-ID"))',
+#'     License = "MIT + file LICENSE",
+#'     Language =  "es"
+#'   )
+#' )
+#' ```
+#'
+#' @param given a character string with the given (first) name of the author.
+#' @param family a character string with the family (last) name of the author.
+#' @param role a character vector specifying the role of the person to be added.
+#' @param email a character string giving an e-mail address of the author to be added.
+#' @param comment a character string providing comments related to the author to be added.
+#'
+#' @export
+#'
+#' @importFrom desc desc_add_author desc_fields desc_get_authors desc_set_authors
+#' @importFrom utils person
+#'
+#' @examples
+#' \dontrun{
+#' # Adds the default Jane Doe to author
+#' use_author()
+#'
+#' # Adds the author information for Ali
+#' use_author(given = "Ali",
+#'           family = "Val",
+#'           role = c("aut", "cre"),
+#'           email = "alval@example.com",
+#'           comment = c(ORCID = "YOUR-ORCID-ID"))
+#'
+#' # Adds a second author Ali2
+#'use_author(given = "Ali2",
+#'           family = "Val2",
+#'           role = c("cph", "fnd"),
+#'           email = "alval2@example.com",
+#'           comment = NULL)
+#' }
+#'
+
+use_author <- function(given = "Jane", family = "Doe", role = c("aut", "cre"), email = "jane@example.com", comment = c(ORCID = "YOUR-ORCID-ID")){
+  # Adapted from use_dependency code and tools provided in the desc package
+  # TODO allow generic authors input like desc_set_authors(authors, file = ".", normalize = FALSE) for multiple author input
+  # TODO create addin to prompt for author information
+  # TODO create a snippet with author information for DESCRIPTION
+  # TODO add tests
+
+  # Set the author as a person
+  author = utils::person(given = given, family = given, role = role, email = email, comment = comment)
+
+  # Obtain the current DESCRIPTION fields
+  current_desc_fields <- desc::desc_fields()
+
+  # If the `Authors@R` is not a current field then add the field and fill in the author provided
+  if(any(current_desc_fields == "Authors@R") == FALSE){
+    ui_done("`Authors@R` field added to DESCRIPTION and filled with {ui_value(author)}.")
+    desc::desc_set_authors(authors = author,
+                           file = proj_get(),
+                           normalize = TRUE)
+    return(invisible())
+  } else if (any(current_desc_fields == "Authors@R") == TRUE){
```

Comment:
If this `if () {...} else if () {...}` stays, note that the `if()` block ends with a return, an "early return". So you don't need `else if() {...}`.  Bring all the code that follows back up to the top-level.

Reply:
Typo on my part. That is what I get for trying to code this while at useR. I've removed these conditionals and made sure to return(invisible only once at the end of the function.